diff --git a/lua/lua-dev/sumneko.lua b/lua/lua-dev/sumneko.lua
index f4a901d..00a8155 100644
--- a/lua/lua-dev/sumneko.lua
+++ b/lua/lua-dev/sumneko.lua
@@ -7,14 +7,14 @@ function M.library(opts)
   local ret = {}
 
   if opts.library.types then
-    ret[M.types()] = true
+    table.insert(ret, M.types())
   end
 
   local function add(lib, filter)
     for _, p in pairs(vim.fn.expand(lib .. "/lua", false, true)) do
       p = vim.loop.fs_realpath(p)
       if p and (not filter or filter[vim.fn.fnamemodify(p, ":h:t")]) then
-        ret[p] = true
+        table.insert(ret, p)
       end
     end
   end
@@ -63,21 +63,7 @@ function M.types()
 end
 
 function M.setup(opts)
-  local ok, lspconfig = pcall(require, "lspconfig")
-
   return {
-    on_new_config = ok and lspconfig.util.add_hook_after(
-      lspconfig.util.default_config.on_new_config,
-      function(config, root_dir)
-        -- remove the root_dir from the workspace, otherwise diagnostics break. See #21
-        -- Should no longer be needed with workspace.nvim
-        local lib = vim.tbl_deep_extend("force", {}, config.settings.Lua.workspace.library)
-        lib[vim.loop.fs_realpath(root_dir) .. "/lua"] = nil
-        lib[vim.loop.fs_realpath(root_dir)] = nil
-        config.settings.Lua.workspace.library = lib
-        return config
-      end
-    ),
     settings = {
       Lua = {
         runtime = {
