From 8a8c8254759d26220d1e2044d52bb104d87e8155 Mon Sep 17 00:00:00 2001
From: Stanislav Asunkin <1353637+stasjok@users.noreply.github.com>
Date: Sun, 19 Oct 2025 13:35:50 +0300
Subject: add ability to customize indexing debouncing via initialization
 options


diff --git a/packages/lsp-server/src/common/startServer.ts b/packages/lsp-server/src/common/startServer.ts
index 674a639..eeb595d 100644
--- a/packages/lsp-server/src/common/startServer.ts
+++ b/packages/lsp-server/src/common/startServer.ts
@@ -171,6 +171,7 @@ export function startServer(
 		documents.onDidOpen(event => symbolIndex.addSyncFile(event.document.uri));
 		// Debounced update to avoid frequent re-index during rapid typing
 		let debouncedUpdateTimer: NodeJS.Timeout | undefined;
+		const debounceTime = params.initializationOptions?.debounceTime || 150
 		documents.onDidChangeContent(event => {
 			symbolIndex.addSyncFile(event.document.uri);
 			if (debouncedUpdateTimer) clearTimeout(debouncedUpdateTimer);
@@ -180,7 +181,7 @@ export function startServer(
 				} catch (e) {
 					serverLogger.debug(`debounced symbolIndex.update error: ${String(e)}`);
 				}
-			}, 150);
+			}, debounceTime);
 		});
 
 		connection.onDidChangeWatchedFiles(e => {
-- 
2.50.1

