From b36ef4830019e640b3c762f8b233d908e2b0a235 Mon Sep 17 00:00:00 2001
From: Stanislav Asunkin <1353637+stasjok@users.noreply.github.com>
Date: Tue, 21 Oct 2025 20:56:40 +0300
Subject: fix: support renaming and finding references for narrations


diff --git a/packages/lsp-server/src/common/features/references.ts b/packages/lsp-server/src/common/features/references.ts
index 66b2f20..cf022c3 100644
--- a/packages/lsp-server/src/common/features/references.ts
+++ b/packages/lsp-server/src/common/features/references.ts
@@ -96,8 +96,8 @@ export class ReferencesFeature {
 		const narrationAtPosition = await positionUtils.getNarrationAtPosition(this.trees, document, params.position);
 		if (narrationAtPosition && references.length === 0) {
 			logger.debug(`Found narration at position: ${narrationAtPosition}`);
-			// We don't have references function for narrations yet,
-			// but could be added in the future
+			// Find all references to this narration
+			references = await this.findNarrationReferences(narrationAtPosition)
 		}
 
 		// Try to find a link at the position
@@ -208,6 +208,30 @@ export class ReferencesFeature {
 		return references;
 	}
 
+	/**
+	 * Find all references to a specific payee
+	 */
+	private async findNarrationReferences(narrationName: string): Promise<lsp.Location[]> {
+		// Find all narration usage locations
+		const narrationUsages = await this.symbolIndex.findAsync({
+			[SymbolKey.TYPE]: SymbolType.NARRATION,
+			name: narrationName,
+		}) as SymbolInfo[];
+
+		const references: lsp.Location[] = [];
+
+		// Convert to LSP Locations
+		narrationUsages.forEach(ref => {
+			references.push({
+				uri: ref._uri,
+				range: getRange(ref),
+			});
+		});
+
+		logger.debug(`Found ${references.length} references to narration: ${narrationName}`);
+		return references;
+	}
+
 	/**
 	 * Find all poptag references to a specific pushtag
 	 */
-- 
2.50.1

