diff --git a/lua/impatient.lua b/lua/impatient.lua
index 0c0ce88..a7de903 100644
--- a/lua/impatient.lua
+++ b/lua/impatient.lua
@@ -34,7 +34,7 @@ local M = {
     cache = {},
     profile = nil,
     dirty = false,
-    path = std_cache..'/luacache_chunks',
+    path = std_cache..'/luacache_chunks'..(_G.__luacache_suffix or ''),
     get = function(self, path)
       return self.cache[modpath_mangle(path)]
     end,
@@ -46,7 +46,7 @@ local M = {
     cache = {},
     profile = nil,
     dirty = false,
-    path = std_cache..'/luacache_modpaths',
+    path = std_cache..'/luacache_modpaths'..(_G.__luacache_suffix or ''),
     get = function(self, mod)
       if self.cache[mod] then
         return modpath_unmangle(self.cache[mod])
