diff --git a/lsp/rust_analyzer.lua b/lsp/rust_analyzer.lua
index 3ac16187..23eccb40 100644
--- a/lsp/rust_analyzer.lua
+++ b/lsp/rust_analyzer.lua
@@ -73,7 +73,7 @@ local function is_library(fname)
 
   local sysroot_src = user_sysroot_src() or default_sysroot_src()
 
-  for _, item in ipairs { toolchains, registry, git_registry, sysroot_src } do
+  for _, item in ipairs { '/nix/store', toolchains, registry, git_registry, sysroot_src } do
     if item and vim.fs.relpath(item, fname) then
       local clients = vim.lsp.get_clients { name = 'rust_analyzer' }
       return #clients > 0 and clients[#clients].config.root_dir or nil
